service/host.go

```diff
@@ -156,7 +156,9 @@ func (s *host) shutdown(err error, reason string, exitCode *int) (bool, error) {
 
 	// Stop the metrics reporting
 	if s.scope != nil {
-		s.scope.Close()
+		if err = s.metricsCloser.Close(); err != nil {
```

Comment:
small possibility for an upstream bug to affect us -- `s.metricsCloser` could be `nil` even if `s.scope` is non-nil. Perhaps we should be defensive and change line 158 to also check if `s.metricsCloser` is non-nil?